contracts/AuctionBase.sol

```diff
@@ -179,37 +197,46 @@ contract AuctionBase is Pausable {
   }
 
   /// @dev Cancels an auction unconditionally.
-  function _cancelAuction(uint256 _id) internal {
-    Auction storage auction = auctions[_id];
-    require(_isActive(auction));
-    require(msg.sender == auction.seller);
+  function _cancelAuction(uint256 _auctionId)
+    internal
+    isExpectedState(AuctionStatus.Ongoing, _auctionId)
+    onlySeller(_auctionId)
+  {
+    Auction storage auction = auctions[_auctionId];
     auction.cancelled = true;
+
     _removeAuction(auction.nftAddress, auction.tokenId);
     _transfer(auction.nftAddress, auction.seller, auction.tokenId);
-    AuctionCancelled(_id, auction.nftAddress, auction.tokenId);
+
+    AuctionCancelled(_auctionId, auction.nftAddress, auction.tokenId);
   }
 
   /// @dev Removes an auction from mapping
   function _removeAuction(address _nft, uint256 _tokenId) internal {
     delete nftToTokenIdToAuctionId[_nft][_tokenId];
   }
 
-  /// @dev Returns true if the NFT is on auction.
-  /// @param _auction - Auction to check.
-  function _isActive(Auction storage _auction) internal view returns (bool) {
-    return (_auction.startedAt > 0 && !_auction.cancelled && _getAuctionEndAt(_auction) <= now);
+  modifier onlySeller(uint256 _auctionId) {
+    Auction memory auction = auctions[_auctionId];
+    require(msg.sender == auction.seller);
+    _;
   }
 
-  function _isEnded(Auction storage _auction) internal view returns (bool) {
-    return (_getAuctionEndAt(_auction) > now);
+  modifier isExpectedState(AuctionStatus _expectedState, uint256 _auctionId) {
+    require(_expectedState == _getAuctionStatus(_auctionId));
+    _;
   }
 
-  function _isWithdrawable(Auction storage _auction) internal view returns (bool) {
-    return (_auction.cancelled || _isEnded(_auction));
-  }
+  function _getAuctionStatus(uint256 _auctionId) internal view returns (AuctionStatus) {
```

Comment:
Try not to mix functions and modifiers. Can you delineate the file into external functions, internal functions and modifiers? I'd tried to do that initially.